stored_requests/multifetcher.go

```diff
@@ -0,0 +1,50 @@
+package stored_requests
+
+import (
+	"context"
+	"encoding/json"
+)
+
+// MultiFetcher is a Fetcher composed of multiple sub-Fetchers that are all polled for results.
+type MultiFetcher []Fetcher
+
+// FetchRequests implements the Fetcher interface for MultiFetcher
+func (mf *MultiFetcher) FetchRequests(ctx context.Context, ids []string) (map[string]json.RawMessage, []error) {
+	var errs []error
+	result := make(map[string]json.RawMessage, len(ids))
+	missingIDs := 0 // The number of missing IDs that each fetcher reported.
+	// If the number of errors == number of missing IDs, then it is likely that the errors are simply due
+	// to the IDs missing from the return result.
+	numIDs := len(ids)
+	// suspect fetchers ... fetchers that returned errors that don't match the number of missing IDs.
+	suspect := 0
+
+	// Loop over the fetchers
+	for _, f := range *mf {
+		remainingIDs := make([]string, 0, len(ids))
+		for _, id := range ids {
+			if _, ok := result[id]; !ok {
+				remainingIDs = append(remainingIDs, id)
+			}
+		}
+		missingIDs = missingIDs + len(remainingIDs)
+		if len(errs) > 0 && len(errs) != len(remainingIDs) {
```

Comment:
There's a corner case here where `len(remainingIDs) == 0` and a single error returned could either be a missing ID or a more serious problem that we may want to return. It may make more sense to define something like this in fetcher.go:

```go
type NotFoundError string

func (e NotFoundError) Error() string {
    return fmt.Sprintf(`Stored Request with ID="%s" not found.`, e)
}
```
And require the fetchers use this error to signify a missing ID so that we can explicitly check at the end:
```go
cleanedErrs := make([]error, 0, len(errs))
for _, err := errs {
  if _, ok := err.(NotFoundError); ok {
    if _, present := result[err]; present {
      // previously missing ID was found, omit error
      continue
    }
  }
  cleanedErrs = append(cleanedErrs, err)
}
```
and then we could remove all of the code around `suspect` and `missingIDs` and just check:
```go
if len(cleanedErrs) > 0 {
  return result, cleanedErrs
}
return result, nil
```

Reply:
I had pondered that, and thought perhaps it should go into its own PR since it involved modifying all the fetchers directly. But if you think it fits inside of this PR I am good with that as well.

Reply:
It does increase the scope and it's not a perfect solution as there's no way to statically ensure that `Fetcher` implementations are always using the error when they should, but I think it may be worth it and safer than assuming `len(errs) > 0 && len(errs) == len(remainingIDs)` is OK.